examples/src/indexes/vector_stores/redis/redis_query.ts

```diff
@@ -0,0 +1,65 @@
+import { createClient } from "redis";
+import { OpenAI } from "langchain/llms/openai";
+import { OpenAIEmbeddings } from "langchain/embeddings/openai";
+import { VectorDBQAChain } from "langchain/chains";
+import { RedisVectorStore } from "langchain/vectorstores/redis";
+
+const client = createClient({
+  url: process.env.REDIS_URL ?? "redis://localhost:6379",
+});
+await client.connect();
+
+const vectorStore = new RedisVectorStore(new OpenAIEmbeddings(), {
+  redisClient: client,
+  indexName: "docs",
+});
+
+/* Simple standalone search in the vector DB */
+const simpleRes = await vectorStore.similaritySearch("redis", 1);
+console.log(simpleRes);
+/*
+[
+  Document {
+    pageContent: "redis is fast",
+    metadata: { foo: "bar" }
+  }
+]
+*/
+
+/* Search in the vector DB using filters */
+const filterRes = await vectorStore.similaritySearch("redis", 3, ["qux"]);
+console.log(filterRes);
+/*
+[
+  Document {
+    pageContent: "consectetur adipiscing elit",
+    metadata: { baz: "qux" },
```

Comment:
How will this metadata filtering work when there are multiple keys in metadata?

Reply:
The implemented filtering is only text-based for now. The filter matches the metadata that contains any of the provided strings in the filters array.
For example, with two records having the following metadata:
1. `metadata: { foo: 'bar', baz: 'qux' }`
2. `metadata: { bar: 'bla' }`

A search with `filter = ['bar']` will return both results because they both have 'bar' in their stringified metadata. It checks for either a value or a key containing that text.

Redis has more complex ways of filtering but this one was the most straightforward to implement and the most useful one imo.
An alternative to allow more complex filtering without overcomplicating the library would be making `filter` a `string` and allow providing a [Redis query syntax string](https://redis.io/docs/stack/search/reference/query_syntax/#a-few-query-examples).

Reply:
Part of the challenge(?) here is that Redis expects you to define the schema declaratively in your index construction. This was by design, and ultimately gives devs more power. But expressing this through LangChain (or any abstraction layer) makes it a bit harder to expose without lot's of additional code. We're ideating this concept on the LC python lib now.

Allowing a user to pass in a Redis query string for more complex metadata filtering would be interesting.. but also still wouldn't handle anything beyond the stringified metadata field right? (redis also supports Tag fields, Numeric fields, in addition to Text).

@nfcampos @mvaker my recommendation would be we start here and then work on a full, unified implementation for metadata filtering with Redis + LangChain across all languages.

Reply:
Thanks for your comment @tylerhutcherson. It’s true that a query string wouldn’t handle anything beyond the stringified metadata field, but ultimately it would let users to use more complex syntax such as negative operator, AND queries, and so on.
Of course, having a full, unified implementation for metadata filtering would be great.
Let’s build up from here.